jmclient/jmclient/configure.py

```diff
@@ -529,13 +586,36 @@ def get_irc_mchannels():
             server_data["socks5_host"] = jm_single().config.get(section, "socks5_host")
             server_data["socks5_port"] = jm_single().config.get(section, "socks5_port")
 
-        for option, otype in req_fields:
+        for option, otype in irc_fields:
             val = jm_single().config.get(section, option)
             server_data[option] = otype(val)
         server_data['btcnet'] = get_network()
         configs.append(server_data)
-    return configs
 
+    # processing the onion sections:
+    for section in sections:
+        if not jm_single().config.has_option(section, "type") or \
+        not jm_single().config.get(section, "type").lower() == "onion":
+            continue
+        onion_data = {}
+        for option, otype in onion_fields:
+            try:
+                val = jm_single().config.get(section, option)
+            except NoOptionError:
+                continue
+            onion_data[option] = otype(val)
+        # the onion messaging section must specify whether
+        # to serve an onion:
+        if mode == "MAKER":
+            onion_data["serving"] = True
+        else:
+            onion_data["serving"] = False
```

Comment:
```suggestion
        onion_data['serving'] = mode == 'MAKER'
```

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1182/commits/09c510742dbf4ffd23502289bfba7f096bd5bd43